dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -25,691 +16,343 @@
+    final static String SELECT_ALL = "select * from multi_tree  ";
+    final static String SELECT_BY_CHILD = "select * from multi_tree where child = ?  order by parent1, parent2, relation_type ";
+    final static String SELECT_BY_PARENTS_AND_RELATIONS =
+            " select * from multi_tree where parent1 = ? and parent2 = ? and relation_type = ? order by tree_order";
 
-			DotConnect db = new DotConnect();
-			db.setSQL("delete from multi_tree where parent1 =? and parent2 = ? and child = ? ");
-			db.addParam(inode1.getInode());
-			db.addParam(inode2.getInode());
-			db.addParam(inode3.getInode());
-			db.getResult();
 
-			updateHTMLPageVersionTS(inode1.getInode());
-			refreshPageInCache(inode1.getInode());
-			
-		}
-		catch (Exception e) {
-			throw new DotRuntimeException(e.getMessage());
-		}
-
-	}
+    public static void deleteMultiTree(final MultiTree mTree) throws DotDataException {
+        _dbDelete(mTree);
+        updateHTMLPageVersionTS(mTree.getHtmlPage());
+        refreshPageInCache(mTree.getHtmlPage());
+    }
 
     /**
-     * Just invoking deleteMultiTreeByParent1 with null language.
-     * @throws DotSecurityException 
+     * Deletes multi-tree relationship given a MultiTree object. It also updates the version_ts of
+     * all versions of the htmlpage passed in (multiTree.parent1)
+     *
+     * @param multiTree
+     * @throws DotDataException
+     * @throws DotSecurityException
      * 
-     * @see MultiTreeFactory#deleteMultiTreeByParent1(Identifier parent, Long
-     *      languageId)
      */
-    public static void deleteMultiTreeByParent1(Contentlet contentlet) throws DotDataException, DotSecurityException {
-        Identifier identifier = new Identifier();
-        identifier.setId(contentlet.getIdentifier());
+    private static void _dbDelete(final MultiTree mTree) throws DotDataException {
+
+        DotConnect db = new DotConnect().setSQL(DELETE_SQL)
+            .addParam(mTree.getHtmlPage())
+            .addParam(mTree.getContainer())
+            .addParam(mTree.getContentlet())
+            .addParam(mTree.getRelationType());
+        db.loadResult();
 
-        deleteMultiTreeByParent1(identifier, contentlet.getLanguageId());
     }
 
     /**
-     * Just invoking deleteMultiTreeByParent1 with null language.
-     * @throws DotSecurityException 
+     * use the relationType for specificity{link
+     * {@link #getMultiTree(String, String, String, String)}
      * 
-     * @see MultiTreeFactory#deleteMultiTreeByParent1(Identifier parent, Long
-     *      languageId)
+     * @param htmlPage
+     * @param container
+     * @param childContent
+     * @return
+     * @throws DotDataException
      */
-    public static void deleteMultiTreeByParent1(Identifier parent) throws DotDataException, DotSecurityException {
-        deleteMultiTreeByParent1(parent, null);
+    @Deprecated
+    public static MultiTree getMultiTree(Identifier htmlPage, Identifier container, Identifier childContent)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)